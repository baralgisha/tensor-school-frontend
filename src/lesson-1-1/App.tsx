function App () {
   return (
      <>
         <h1>Топ песен группы "Любэ"</h1>
         <h2>(кто это вообще?)</h2>
         <ul>
            <li><a href="vk.com">vk</a></li>
            <li><a href="vk.com">vk</a></li>
            <li><a href="vk.com">vk</a></li>
         </ul>
      </>
   )
}

export default App