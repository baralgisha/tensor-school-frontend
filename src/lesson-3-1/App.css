html, body, #root {
   height: 100%;
   margin: 0;
}

.flex-container {
   display: flex;
   justify-content: space-between;
   align-items: center;
   width: 700px;
   height: 300px;
}

.flex-container > div {
   width: 100px;
   height: 100px;
   background-color: #DAA520;
   border-radius: 50%;
}

.flex-container :first-child,
.flex-container :last-child {
   align-self: flex-start;
}

.flex-container :nth-child(3) {
   align-self: flex-end;
}
