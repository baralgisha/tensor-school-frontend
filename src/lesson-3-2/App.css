.grid-container {
   display: grid;
   grid-template-columns: repeat(2, 1fr);
   grid-template-rows: 1fr 200px 1fr;
   justify-content: stretch;
   align-items: stretch;
   gap: 20px;
   height: 400px;
   width: 700px;
}

.grid-container > div {
   background-color: #DAA520;
}
