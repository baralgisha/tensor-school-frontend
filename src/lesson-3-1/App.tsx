import './App.css';

function App() {
   return (
   <>
      <section className="flex-container">
         <div></div>
         <div></div>
         <div></div>
         <div></div>
         <div></div>
      </section>
   </>)
}

export default App;