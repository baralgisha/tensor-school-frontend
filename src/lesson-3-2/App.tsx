import './App.css';

function App() {
   return (
   <>
      <section className="grid-container">
         <div></div>
         <div></div>
         <div></div>
         <div></div>
         <div></div>
      </section>
   </>)
}

export default App;