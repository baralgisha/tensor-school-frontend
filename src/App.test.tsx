import {render, fireEvent} from "@testing-library/react";
import App from './App';

describe('toggle list', () => {
    it('check unordered list', () => {
        const {container} = render(<App/>);

        expect(container).toMatchSnapshot();
    })

    it('check ordered list', () => {
        const {queryByTestId, container} = render(<App/>);

        const listTypeToggle = queryByTestId('listType') as HTMLElement;
        fireEvent.click(listTypeToggle);

        expect(container).toMatchSnapshot();
    })
})

describe('toggle font size', () => {
    it('check small font', () => {
        const {queryByTestId, container} = render(<App/>);

        const fontToggle = queryByTestId('fontSize') as HTMLElement;
        fireEvent.click(fontToggle);

        expect(container).toMatchSnapshot();
    })
})

describe('toggle color', () => {
    it('check violet', () => {
        const {queryByTestId, container} = render(<App/>);

        const colorToggle = queryByTestId('color') as HTMLElement;
        fireEvent.click(colorToggle);

        expect(container).toMatchSnapshot();
    })
})
