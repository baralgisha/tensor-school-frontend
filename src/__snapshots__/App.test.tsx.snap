// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle color check violet 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="App__list App__list-big"
      data-testid="list"
      style="color: violet;"
    >
      <ul>
        <li>
          Посмотреть вокруг
        </li>
        <li>
          Оглянуться назад
        </li>
        <li>
          Рыбов купить
        </li>
        <li>
          "Красивое" сказать
        </li>
      </ul>
    </div>
    <div
      class="App__buttons"
      data-testid="buttons"
    >
      <button
        data-testid="listType"
      >
        Хочу цифры
      </button>
      <button
        data-testid="color"
      >
        Хочу красный
      </button>
      <button
        data-testid="fontSize"
      >
        Хочу маленькие
      </button>
    </div>
    <div
      class="App__image"
    />
  </div>
</div>
`;

exports[`toggle font size check small font 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="App__list App__list-small"
      data-testid="list"
      style="color: red;"
    >
      <ul>
        <li>
          Посмотреть вокруг
        </li>
        <li>
          Оглянуться назад
        </li>
        <li>
          Рыбов купить
        </li>
        <li>
          "Красивое" сказать
        </li>
      </ul>
    </div>
    <div
      class="App__buttons"
      data-testid="buttons"
    >
      <button
        data-testid="listType"
      >
        Хочу цифры
      </button>
      <button
        data-testid="color"
      >
        Хочу фиолетовый
      </button>
      <button
        data-testid="fontSize"
      >
        Хочу большие
      </button>
    </div>
    <div
      class="App__image"
    />
  </div>
</div>
`;

exports[`toggle list check ordered list 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="App__list App__list-big"
      data-testid="list"
      style="color: red;"
    >
      <ol>
        <li>
          Посмотреть вокруг
        </li>
        <li>
          Оглянуться назад
        </li>
        <li>
          Рыбов купить
        </li>
        <li>
          "Красивое" сказать
        </li>
      </ol>
    </div>
    <div
      class="App__buttons"
      data-testid="buttons"
    >
      <button
        data-testid="listType"
      >
        Хочу точки
      </button>
      <button
        data-testid="color"
      >
        Хочу фиолетовый
      </button>
      <button
        data-testid="fontSize"
      >
        Хочу маленькие
      </button>
    </div>
    <div
      class="App__image"
    />
  </div>
</div>
`;

exports[`toggle list check unordered list 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="App__list App__list-big"
      data-testid="list"
      style="color: red;"
    >
      <ul>
        <li>
          Посмотреть вокруг
        </li>
        <li>
          Оглянуться назад
        </li>
        <li>
          Рыбов купить
        </li>
        <li>
          "Красивое" сказать
        </li>
      </ul>
    </div>
    <div
      class="App__buttons"
      data-testid="buttons"
    >
      <button
        data-testid="listType"
      >
        Хочу цифры
      </button>
      <button
        data-testid="color"
      >
        Хочу фиолетовый
      </button>
      <button
        data-testid="fontSize"
      >
        Хочу маленькие
      </button>
    </div>
    <div
      class="App__image"
    />
  </div>
</div>
`;
